src/Generator.cpp

```diff
@@ -21,362 +26,655 @@
+                y_c = float(c.getY());
+                float cell_radius = sqrt((x_c*x_c)+(y_c-y_c));
+
+                if (cell_radius <= float(parameters_.geometry().limit_first_zone))
+                    side = parameters_.geometry().small_cell_side;
+                else
+                    side = parameters_.geometry().large_cell_side;
+
+                Point* cornerA = new Point(x_c - float(side*sqrt(3)/2), y_c + float(side));
+                Point* cornerB = new Point(x_c + float(side*sqrt(3)/2), y_c + float(side));
+                Point* cornerC = new Point(x_c + float(side*sqrt(3)/2), y_c - float(side));
+                Point* cornerD = new Point(x_c - float(side*sqrt(3)/2), y_c - float(side));
+
+                std::set<Tree*> alreadyAdded;
 
-  // Exporting histograms to file
-  //hEnergyGen.Write();
-  //hTransverseProfile.Write();
-  //hPhiProfile.Write();
-  //hSpotEnergy.Write();
-  //hCellEnergyDist.Write();
-  //hEnergySum.Write();
+                Tree* leafA = tree->getLeaf(cornerA);
+                leafA->addCell(&c);
+                alreadyAdded.insert(leafA);
 
-  //for (const auto& id_hist : hCellEnergyMap) { 
-    //id_hist.second.Write();
-  //}  
+                Tree* leafB = tree->getLeaf(cornerB);
 
+                if(alreadyAdded.count(leafB) == 0) {
+                    leafB->addCell(&c);
+                    alreadyAdded.insert(leafB);
+                }
 
-  t.Stop();
-  t.Print();
+                Tree* leafC = tree->getLeaf(cornerC);
 
-  canvas.emplace_back(display(hCellEnergyMap));    
+                if(alreadyAdded.count(leafC) == 0) {
+                    leafC->addCell(&c);
+                    alreadyAdded.insert(leafC);
+                }
 
-  // Writing energy map plots
-  for(const auto& canvas_ptr : canvas) {
-    canvas_ptr->Write();
-  }
+                Tree* leafD = tree->getLeaf(cornerD);
 
+                if(alreadyAdded.count(leafD) == 0) {
+                    leafD->addCell(&c);
+                    alreadyAdded.insert(leafD);
+                }
+            }
+
+            tree_map.insert({layer_id, tree});
+        }
+    }
+
+    int thick = 0;
+
+    for (unsigned iev=1; iev <= nevents; iev++) {
+        // cout << "================ Simulating event: " << iev << " ================" << endl;
+
+        // initialize event
+        Event event(0, iev); // default run number =0
+
+        energygen = 0.;
+        energygenincells = 0.;
+        energyrec = 0.;
+
+        // Particles generation
+        unsigned npart = parameters_.general().part_type.size();
+        event.setnPart(npart);
+
+
+        for (int ip : parameters_.general().part_type) {
+
+            // particle incidents parameters
+            double eta_incident, phi_incident;
+            double energy_incident;
+            if (parameters_.generation().eta_fluctuation)
+                eta_incident = gun_.Uniform(parameters_.generation().eta_range_min, parameters_.generation().eta_range_max);
```

Comment:
Here and after: better to add brackets around the `if` `else`

Reply:
Fixed